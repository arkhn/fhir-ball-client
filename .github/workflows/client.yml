name: pyrog-client

on: [push]

jobs:
  build:
    name: Build Pyrog client
    runs-on: ubuntu-18.04

    strategy:
      matrix:
        node-version: [12.x]

    steps:
      - uses: actions/checkout@v1

      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}

      - name: Cache node modules
        uses: actions/cache@v1
        with:
          path: client/node_modules
          key: ${{ runner.OS }}-${{ matrix.node-version}}-node-modules-${{ hashFiles('client/yarn.lock') }}

      - name: Install Dependencies
        run: |
          cd client
          yarn install

      - name: Build
        run: |
          cd client
          yarn build

  # Same cleaning need to be done, see Issue #105
  #      - name: Lint
  #        run: |
  #          cd client
  #          yarn lint

  publish:
    name: Build and publish
    env:
      GITHUB_REGISTRY: docker.pkg.github.com
      IMAGE: arkhn/pyrog-client
    runs-on: ubuntu-18.04
    needs: build
    if: github.ref == 'refs/heads/master'
    steps:
      - uses: actions/checkout@v1

      - name: Build
        run: |
          cd client/
          yarn docker:build:staging

      - name: Tag docker images
        run: |
          docker tag $IMAGE $IMAGE:$GITHUB_SHA
          docker tag $IMAGE $GITHUB_REGISTRY/${{ github.repository }}/pyrog-client:latest
          docker tag $IMAGE $GITHUB_REGISTRY/${{ github.repository }}/pyrog-client:$GITHUB_SHA

      - name: Publish to the docker hub
        run: |
          echo "${{ secrets.DOCKER_PASSWORD }}" |  docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
          docker push $IMAGE:latest
          docker push $IMAGE:$GITHUB_SHA

      - name: Publish to Github Packages
        run: |
          echo "${{ secrets.GITHUB_TOKEN }}" | docker login $GITHUB_REGISTRY -u ${{ github.actor }} --password-stdin
          docker push $GITHUB_REGISTRY/${{ github.repository }}/pyrog-client:latest
          docker push $GITHUB_REGISTRY/${{ github.repository }}/pyrog-client:$GITHUB_SHA
