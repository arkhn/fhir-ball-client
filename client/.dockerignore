# ignore everything
**/*

# except
!.dockerignore
!Dockerfile
!yarn.lock
!package.json
!public
!src
!tsconfig.json
!tsconfig.build.json
!.env*
!build