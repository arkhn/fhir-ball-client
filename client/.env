PORT=1729
REACT_APP_HTTP_BACKEND_URL=http://localhost:1000
REACT_APP_RIVER_URL=https://staging.arkhn.com/river
REACT_APP_PAGAI_URL=https://staging.arkhn.com/pagai
REACT_APP_CLEANING_SCRIPTS_URL=https://staging.arkhn.com/scripts
REACT_APP_FHIR_API_URL=https://staging.arkhn.com/api
