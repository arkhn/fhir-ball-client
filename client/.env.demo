PORT=1729
REACT_APP_HTTP_BACKEND_URL=https://fhir.demo.arkhn.com/pyrog
REACT_APP_FHIRPIPE_URL=https://fhir.demo.arkhn.com/pipe
REACT_APP_CLEANING_SCRIPTS_URL=https://fhir.demo.arkhn.com/scripts
REACT_APP_FHIR_API_URL=https://fhir.demo.arkhn.com/api
REACT_APP_PAGAI_URL=https://fhir.demo.arkhn.com/pagai
