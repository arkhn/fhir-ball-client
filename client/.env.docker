#pyrog-client
PORT=1729
REACT_APP_HTTP_BACKEND_URL=http://pyrog-server:1000
REACT_APP_FHIRPIPE_URL=http://localhost:3000
REACT_APP_CLEANING_SCRIPTS_URL=http://cleaning-scripts:5000
REACT_APP_PAGAI_URL=http://pagai:4000

#cleaning-scripts
GITHUB_TOKEN=<token>


#pagai
PYROG_URL=http://pyrog-server:4000
PYROG_TOKEN=<token>
