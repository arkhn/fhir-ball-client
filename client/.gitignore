node_modules/
dist/*
build/

coverage
yarn-error.log

.DS_Store