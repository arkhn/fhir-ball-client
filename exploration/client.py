import os
import psycopg2
import yaml

configFileName = './conf.custom.yml' if os.path.exists('./conf.custom.yml') else './conf.default.yml'

def main():
    """Initiates connection with distant postgres database."""

    with open(configFileName) as config_file:
        try:
            # Load config
            config = yaml.safe_load(config_file)
            connection = None

            try:
                # Initialise client
                connection = psycopg2.connect(
                    host=config['db_host'],
                    port=config['db_port'],
                    database=config['db_name'],
                    user=config['db_user'],
                    password=config['db_password']
                )

                # Run SQL query
                cursor = connection.cursor()
                cursor.execute("SELECT * from patients limit 10;")
                data = cursor.fetchall()
                print(data)

            except (Exception, psycopg2.Error) as error :
                print ("Error while connecting to PostgreSQL", error)
            finally:
                # Closing database connection.
                if(connection):
                    cursor.close()
                    connection.close()
                    print("PostgreSQL connection is closed")

        except yaml.YAMLError as error:
            print(error)

if __name__ == "__main__":
    main()