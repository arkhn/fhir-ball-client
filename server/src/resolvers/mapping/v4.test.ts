import { PrismaClient } from '@prisma/client'

import importMappingV4 from './v4'
import * as mappingV4 from '../../../test/fixtures/chimio-mapping-v4.json'

const mockCreateResource = jest.fn()
jest.mock('@prisma/client', () => ({
  PrismaClient: jest.fn().mockImplementation(() => ({
    resource: {
      create: (data: any) => mockCreateResource(data),
    },
  })),
}))

describe('import mapping V4', () => {
  const sourceId = '01234567'
  const resourceCount = 2
  const { resources } = mappingV4 as any

  beforeEach(() => {
    mockCreateResource.mockClear()
  })

  it('should send a query per resource', async () => {
    await importMappingV4(new PrismaClient(), sourceId, resources)
    expect(mockCreateResource).toHaveBeenCalledTimes(resourceCount)
    expect(mockCreateResource.mock.calls[0]).toMatchSnapshot() // EpisodeOfCare - HopitalStay
    expect(mockCreateResource.mock.calls[1]).toMatchSnapshot() // Patient
  })

  it('should have cleaned the resource and attributes', async () => {
    await importMappingV4(new PrismaClient(), sourceId, resources)
    expect(mockCreateResource.mock.calls[0][0]).toEqual({
      data: {
        id: resources[0].id,
        label: resources[0].label,
        primaryKeyOwner: resources[0].primaryKeyOwner,
        primaryKeyTable: resources[0].primaryKeyTable,
        primaryKeyColumn: resources[0].primaryKeyColumn,
        source: {
          connect: {
            id: sourceId,
          },
        },
        definitionId: 'EpisodeOfCare',
        attributes: {
          create: expect.arrayContaining([
            {
              id: expect.any(String),
              path: 'period.start',
              definitionId: '',
              mergingScript: 'merge_concat',
              inputs: {
                create: expect.any(Array),
              },
            },
            {
              id: expect.any(String),
              path: 'managingOrganization.reference',
              comments: {
                create: {
                  author: {
                    connect: {
                      email: 'admin@arkhn.com',
                    },
                  },
                  content: 'test',
                },
              },
              definitionId: '',
              mergingScript: null,
              inputs: {
                create: expect.any(Array),
              },
            },
          ]),
        },
      },
    })
  })
})
