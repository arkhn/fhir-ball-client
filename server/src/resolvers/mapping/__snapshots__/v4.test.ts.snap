// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import mapping V4 should send a query per resource 1`] = `
Array [
  Object {
    "data": Object {
      "attributes": Object {
        "create": Array [
          Object {
            "comments": Object {
              "create": Object {
                "author": Object {
                  "connect": Object {
                    "email": "admin@arkhn.com",
                  },
                },
                "content": "test",
              },
            },
            "definitionId": "",
            "id": "ck74z6s2o0005s3utow77alh6",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s2r0007s3utxzvrdfv9",
                  "script": null,
                  "staticValue": "internal",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.reference",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s2u0009s3utu92qsv58",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s2w0011s3utlchaktkd",
                  "script": null,
                  "staticValue": "/Organization/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.system",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s2y0013s3ut92hx0cht",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s330018s3utjuodqpw6",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SERVICE",
                      "id": "ck74z6s310015s3ut24kka7n3",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.value",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s350020s3utonyqbwkj",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3a0024s3utxargsip8",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ETATSEJ",
                      "id": "ck74z6s380022s3utvedq54t8",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "status",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s3c0026s3utstqaga9u",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3g0030s3utlkwrfixv",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SECLEUNIK",
                      "id": "ck74z6s3e0028s3ut5psjydwq",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].value",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s3i0032s3utr5dmi7fq",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3n0036s3utgsb2ffho",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADMISSION",
                      "id": "ck74z6s3l0034s3ut8ilts58r",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3r0040s3utccmlsuwa",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADMISSION2",
                      "id": "ck74z6s3p0038s3ut9idxyft4",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "select_first_not_empty",
            "path": "identifier[1].value",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s3t0042s3ut5xo9nh0x",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3y0046s3utrbo1byee",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "TWCLEUNIK",
                      "id": "ck74z6s3w0044s3ute3w7z6ds",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "type[0].coding[0].code",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s400049s3utivcmb7vo",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4b0059s3utd254bust",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "LIBELTW",
                      "id": "ck74z6s420051s3utubfvs13s",
                      "joins": Object {
                        "create": Array [
                          Object {
                            "id": "ck74z6s440052s3utk0dunr5b",
                            "tables": Object {
                              "create": Array [
                                Object {
                                  "column": "TWCLEUNIK",
                                  "id": "ck74z6s470055s3utuw32fadc",
                                  "owner": "OPS$ACHQ1ABC",
                                  "table": "SEJOUR",
                                },
                                Object {
                                  "column": "TWCLEUNIK",
                                  "id": "ck74z6s490057s3utqo816057",
                                  "owner": "OPS$ACHQ1ABC",
                                  "table": "TYPESEJ",
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "owner": "OPS$ACHQ1ABC",
                      "table": "TYPESEJ",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "type[0].coding[0].display",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s4d0061s3utbbxfls8b",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4h0065s3utonhdfiah",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEOUT",
                      "id": "ck74z6s4f0063s3utjjgh0sk4",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4l0069s3uthpzkygwi",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "HEUREOUT",
                      "id": "ck74z6s4j0067s3ut4gfmb2hr",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "period.end",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s4n0071s3ut75r6unij",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4r0075s3utpttakhte",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEIN",
                      "id": "ck74z6s4p0073s3utgrvzky12",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4v0079s3utitey8m4b",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "HEUREIN",
                      "id": "ck74z6s4t0077s3utkxbdohz9",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "period.start",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s4x0081s3utsqha4wcf",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4z0083s3utfjdcwxjw",
                  "script": null,
                  "staticValue": "internal",
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.reference",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s510085s3uto5vtu3ns",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s530087s3ut0fxdd2hg",
                  "script": null,
                  "staticValue": "/Patient/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.identifier.system",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s560089s3utkfo9kdac",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s590093s3ut4yzvbz1l",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PACLEUNIK",
                      "id": "ck74z6s570091s3utwj5qsnrm",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.identifier.value",
          },
        ],
      },
      "definitionId": "EpisodeOfCare",
      "id": "ck74z6s2l0003s3utqz3gjj5d",
      "label": "Hospital Stay",
      "primaryKeyColumn": "SECLEUNIK",
      "primaryKeyOwner": "OPS$ACHQ1ABC",
      "primaryKeyTable": "SEJOUR",
      "source": Object {
        "connect": Object {
          "id": "01234567",
        },
      },
    },
  },
]
`;

exports[`import mapping V4 should send a query per resource 2`] = `
Array [
  Object {
    "data": Object {
      "attributes": Object {
        "create": Array [
          Object {
            "definitionId": "",
            "id": "ck74z6s2o0006s3utvs7dpyyg",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s2u0010s3ut0jjhi9p9",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "INACTIVPAT",
                      "id": "ck74z6s2r0008s3ut09vzvnxz",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "active",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s2w0012s3utcv9f5yx4",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s310016s3utm4ec00bf",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR1PAT",
                      "id": "ck74z6s2y0014s3utj262dt8x",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s350019s3ut6nhj9eyo",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR2PAT",
                      "id": "ck74z6s330017s3ut6svkgnl3",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s390023s3utl5iu29zc",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR3PAT",
                      "id": "ck74z6s370021s3utbw3d3yfl",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "address[0].line[0]",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s3b0025s3utqcwwbc8q",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3f0029s3utvpsizaxf",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "VILLEPAT",
                      "id": "ck74z6s3d0027s3utvxktapbv",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "address[0].city",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s3i0031s3utlcbb7zwy",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3m0035s3utjinwsvbs",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "CPOSTPAT",
                      "id": "ck74z6s3k0033s3utki6j4rq6",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "address[0].postalCode",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s3o0037s3utn088gz0v",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3s0041s3utzurqvuok",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "INACTIVPAT",
                      "id": "ck74z6s3q0039s3utp7wiulo1",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s3w0045s3utqwindfws",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "IDMOTIFINAC",
                      "id": "ck74z6s3u0043s3utxxv9ieu5",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "deceasedBoolean",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s3y0047s3ut41wvh4np",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s420050s3utkjdzdyn7",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEINACPAT",
                      "id": "ck74z6s400048s3utkolz8hfy",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "deceasedDateTime",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s440053s3utonbxenbm",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s480056s3utiy8062ct",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "TELPAT",
                      "id": "ck74z6s460054s3utze9qa2uh",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "telecom[0].value",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s4a0058s3ut6za31w82",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4c0060s3uta66ykb2f",
                  "script": null,
                  "staticValue": "official",
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].use",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s4e0062s3utmkpa5hdf",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4i0066s3utft9tqgf5",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOMPAT",
                      "id": "ck74z6s4g0064s3utqrlnww9a",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].family",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s4k0068s3utjct66zcq",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4o0072s3utripc836e",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PRENOMPAT",
                      "id": "ck74z6s4m0070s3utf4pze59l",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].given[0]",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s4q0074s3utypb6tw5y",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4s0076s3utnyglsfnk",
                  "script": null,
                  "staticValue": "maiden",
                },
              ],
            },
            "mergingScript": null,
            "path": "name[1].use",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s4u0078s3ut5lotec8n",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s4y0082s3uts7dlg5k7",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOMJFPAT",
                      "id": "ck74z6s4w0080s3ut38d03www",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[1].family",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s500084s3utas3vn3w7",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s540088s3utcos2gzf7",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATENAISPA",
                      "id": "ck74z6s520086s3utq1na3m11",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "birthDate",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s560090s3utq446lo7l",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s590094s3utyh8atj34",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SEXEPAT",
                      "id": "ck74z6s580092s3utkee8vzy1",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "gender",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s5c0095s3ut7vmxicp1",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s5e0096s3uttfo66eca",
                  "script": null,
                  "staticValue": "http://terminology.interopsante.org/CodeSystem/v2-0203",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].system",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s5g0097s3ut9zsjs21s",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s5i0098s3utx3hwyuez",
                  "script": null,
                  "staticValue": "PI",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].code",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s5k0099s3ut3pb37g4a",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s5m0100s3utgstvn3m9",
                  "script": null,
                  "staticValue": "Patient internal identifier",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].display",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s5n0101s3ut2co2emx1",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s5p0102s3ut7b7xbz2d",
                  "script": null,
                  "staticValue": "false",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].userSelected",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s5r0103s3utxqwqh928",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s5v0105s3utv06b4yp7",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOOBSPAT",
                      "id": "ck74z6s5t0104s3utbbf02nna",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].value",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s5x0106s3utr1ce0qov",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s610108s3ut52ay4oir",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PACLEUNIK",
                      "id": "ck74z6s5z0107s3uthh8tx5wd",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[1].value",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s630109s3utbb12yx8c",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s670111s3uts1c1575v",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DOSSIERPAT",
                      "id": "ck74z6s650110s3utxf0lfw5s",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].value",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s690112s3utt0jc7inx",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s6d0114s3utew6s5t2y",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEINPAT",
                      "id": "ck74z6s6b0113s3ut8u2k3iot",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].period.start",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s6f0115s3uth9zxnbwl",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s6j0117s3utg11tbzs8",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEOUTPAT",
                      "id": "ck74z6s6h0116s3utch62zrvp",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].period.end",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s6l0118s3utqol2rbuh",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s6n0119s3utoopvxuuf",
                  "script": null,
                  "staticValue": "/Organization",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.type",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s6p0120s3utizcwysug",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s6r0121s3ut9qzuexx0",
                  "script": null,
                  "staticValue": "/Organization/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.system",
          },
          Object {
            "definitionId": "",
            "id": "ck74z6s6t0122s3utdatirwmg",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck74z6s6x0124s3utj7krrecc",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "CODESERV",
                      "id": "ck74z6s6v0123s3ut7nm8hc2m",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.value",
          },
        ],
      },
      "definitionId": "Patient",
      "id": "ck74z6s2l0004s3ut2os7csti",
      "label": "",
      "primaryKeyColumn": "PACLEUNIK",
      "primaryKeyOwner": "OPS$ACHQ1ABC",
      "primaryKeyTable": "PATIENT",
      "source": Object {
        "connect": Object {
          "id": "01234567",
        },
      },
    },
  },
]
`;
